lib/compiler/src/engine/artifact.rs

```diff
@@ -56,6 +56,17 @@ impl Artifact {
         data: &[u8],
         tunables: &dyn Tunables,
     ) -> Result<Self, CompileError> {
+        let artifact = Self::get_artifact_build(engine, data, tunables)?;
+        let mut inner_engine = engine.inner_mut();
+        Self::from_parts(&mut inner_engine, artifact)
+    }
+
+    #[cfg(feature = "compiler")]
+    fn get_artifact_build(
+        engine: &Engine,
+        data: &[u8],
+        tunables: &dyn Tunables,
+    ) -> Result<ArtifactBuild, CompileError> {
```

Comment:
We can undo this

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3483/commits/b26b48794cc7b00bb5f939dfdf40bba94e9ba86c